docs/x509.rst

```diff
@@ -273,6 +273,49 @@ X.509 Certificate Object
 
         The dotted string value of the OID (e.g. ``"2.5.4.3"``)
 
+X.509 Extensions
+~~~~~~~~~~~~~~~~
+
+.. class:: Extension
+
+    .. versionadded:: 0.9
+
+    All X.509 extensions are registered against this interface.
+
+    .. attribute:: critical
+
+        :type: bool
+
+        Determines whether a given extension is critical or not.
```

Comment:
Maybe let's do that to save duplicating that validation.
On Mar 22, 2015 1:59 PM, "Paul Kehrer" <notifications@github.com> wrote:

> In docs/x509.rst
> <https://github.com/pyca/cryptography/pull/1768#discussion_r26905375>:
>
> > @@ -273,6 +273,49 @@ X.509 Certificate Object
> >
> >          The dotted string value of the OID (e.g. ``"2.5.4.3"``)
> >
> > +X.509 Extensions
> > +~~~~~~~~~~~~~~~~
> > +
> > +.. class:: Extension
> > +
> > +    .. versionadded:: 0.9
> > +
> > +    All X.509 extensions are registered against this interface.
> > +
> > +    .. attribute:: critical
> > +
> > +        :type: bool
> > +
> > +        Determines whether a given extension is critical or not.
>
> Composition is more accurate. This is what an X509 extension really looks
> like:
>
>
>    Extension  ::=  SEQUENCE  {
>         extnID      OBJECT IDENTIFIER,
>         critical    BOOLEAN DEFAULT FALSE,
>         extnValue   OCTET STRING
>                     -- contains the DER encoding of an ASN.1 value
>                     -- corresponding to the extension type identified
>                     -- by extnID
>         }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1768/files#r26905375>.
>